drainer/schema.go

```diff
@@ -173,8 +185,33 @@ func (s *Schema) DropTable(id int64) (string, error) {
 	return table.Name.O, nil
 }
 
+func (s *Schema) appendTableInfo(schemaVersion int64, table *model.TableInfo) {
+	tbls := s.tables[table.ID]
+	tbls = append(tbls, schemaVersionTableInfo{SchemaVersion: schemaVersion, TableInfo: table})
+	if len(tbls) > 2 {
+		tbls = tbls[len(tbls)-2:]
```

Comment:
Does _amend_pessimistic_txn_ add only one additional schema version at most?

Reply:
um..I think we can save the limit version number in a time range like 1hour or 1day?